packages/formik/src/hooks/useOptimizedSelector.ts

```diff
@@ -0,0 +1,28 @@
+import { useMemo } from 'react';
+
+export type Selector<Value, Return> = (value: Value) => Return;
+export type Comparer<Return> = (prev: Return, next: Return) => boolean;
+
+const UNINITIALIZED_VALUE = Symbol();
+
```

Comment:
Add comment

Reply:
I've moved this package to its own unit tested npm package in another PR. see #3124.